teuthology/task/ansible.py

```diff
@@ -0,0 +1,225 @@
+import logging
+import requests
+import os
+import pexpect
+import yaml
+
+from cStringIO import StringIO
+from tempfile import NamedTemporaryFile
+
+from teuthology.exceptions import CommandFailedError
+from teuthology.repo_utils import fetch_repo
+
+from . import Task
+
+log = logging.getLogger(__name__)
+
+
+class LoggerFile(object):
```

Comment:
Well of course ansible itself doesn't know about python loggers. Normally what one would probably do is use ``subprocess.Popen`` to run the process, tell it to pipe stderr and stdout to a buffer, then call ``readlines()`` on the buffer in a loop and log the output. For whatever reason that doesn't work with ``ansible-playbook``; perhaps it writes directly to the terminal instead of using stdout/stderr.

``pexpect`` handles cases like this, and it takes a ``logfile`` argument, so I built a logger with a ``file``-like interface.